### ISSUE TITLE
Cannot read property 'write' of null

### ISSUE THREAD

**Issue by fupps** [evanshortiss/express-expeditious#22]
I am getting following random error. Some times it works for 10 minutes - sometimes it crashes after several seconds. Any hints?
`...\node_modules\express-expeditious\lib\middleware.js:254
        _write = res.socket.write.bind(res.socket)
                            ^

TypeError: Cannot read property 'write' of null
    at processRequest (I:\Projekt\BDEDash\node_modules\express-expeditious\lib\middleware.js:254:29)
    at onCacheReturned (I:\Projekt\BDEDash\node_modules\express-expeditious\lib\middleware.js:89:9)
    at I:\Projekt\BDEDash\node_modules\safejson\src\index.js:55:14
    at deferred (I:\Projekt\BDEDash\node_modules\safejson\src\index.js:15:5)
    at Object.exports.parse (I:\Projekt\BDEDash\node_modules\safejson\src\index.js:51:3)
    at onGetCallComplete (I:\Projekt\BDEDash\node_modules\expeditious\lib\handlers.js:33:18)
    at Immediate.<anonymous> (I:\Projekt\BDEDash\node_modules\expeditious-engine-memory\expeditious-memory-engine.js:50:9)
    at runCallback (timers.js:696:18)
    at tryOnImmediate (timers.js:667:5)
    at processImmediate (timers.js:649:5)
[nodemon] app crashed - waiting for file changes before starting...`

**Comment by fupps** [comment]
With DEBUG on:

`2018-07-26T11:29:48.246Z express-expeditious GET /api/uebersicht/mess/05.06.2018%2011:25:20 - etag did not match. returning cached data.
2018-07-26T11:29:48.247Z express-expeditious GET /api/anlage/fahrt/39/05.06.2018%2011:25:20/120 - checking cache for request using key "GET-/api/anlage/fahrt/39/05.06.2018%2011:25:20/120"
2018-07-26T11:29:48.247Z express-expeditious GET /api/anlage/fahrt/39/05.06.2018%2011:25:20/120 - cache hit. responding with cached data {"headers":"HTTP/1.1 200 OK\r\nX-Powered-By: Express\r\nAccess-Control-Allow-Origin: *\r\nAccess-Control-Allow-Methods: GET, POST, OPTIONS, PUT, PATCH, DELETE\r\nAccess-Control-Allow-Headers: Access-Control-Allow-Headers, Origin,Accept, X-Requested-With, Content-Type, Access-Control-Request-Method, Access-Control-Request-Headers\r\nx-expeditious-cache: hit\r\nContent-Type: application/json; charset=utf-8\r\nETag: W/\"ad64-SxCct2OpY7wLAfweZY68IVOEa78\"\r\nVary: Accept-Encoding\r\nContent-Encoding: gzip\r\nDate: Thu, 26 Jul 2018 11:29:46 GMT\r\nConnection: keep-alive\r\nTransfer-Encoding: chunked","data":{"type":"Buffer","data":[13,10,13,10,97,13,10,31,139,8,0,0,0,0,0,0,10,13,10,52,48,57,13,10,237,220,193,106,219,80,16,133,225,119,241,58,41,51,119,174,164,43,237,76,226,38,1,55,139,198,217,180,148,62,72,233,187,87,178,67,22,229,206,129,66,78,154,197,1,47,162,137,145,205,183,72,252,227,225,126,255,181,219,63,30,247,119,135,221,178,139,121,119,181,123,186,185,127,184,185,63,173,151,251,245,234,118,127,122,254,178,254,92,204,219,181,141,215,86,79,165,44,102,235,227,147,153,125,91,159,242,237,240,176,61,219,125,41,117,9,95,39,151,27,254,60,238,15,207,159,215,95,249,235,228,233,116,216,238,108,87,187,187,175,207,143,183,235,245,195,241,248,116,218,63,222,90,119,234,221,105,233,78,163,59,173,221,233,208,157,142,221,233,212,157,182,238,116,62,79,159,94,68,94,205,134,147,205,23,156,139,217,239,171,183,68,15,161,231,232,65,66,47,66,207,209,11,9,221,133,158,163,59,9,221,132,158,163,27,7,221,219,82,103,161,247,209,207,56,12,244,73,232,57,250,68,66,31,133,158,163,143,36,244,65,232,57,250,64,67,143,14,186,253,141,238,31,0,189,243,30,232,232,65,65,175,75,105,66,79,208,55,28,6,122,8,61,71,15,18,122,17,122,142,94,72,232,46,244,28,221,73,232,38,244,28,221,56,232,231,191,92,250,200,216,71,63,227,48,208,155,208,115,244,70,66,159,132,158,163,79,36,244,81,232,57,250,72,66,31,150,82,244,143,52,65,223,112,24,232,85,232,57,122,37,161,135,208,115,244,32,161,23,161,231,232,133,132,238,66,207,209,157,132,110,66,207,209,141,129,110,203,48,11,61,65,111,23,28,6,122,19,122,142,222,72,232,147,208,115,244,137,132,62,106,239,37,71,31,25,123,47,43,250,32,244,28,125,32,161,107,107,23,160,83,182,118,87,116,109,237,2,116,202,214,238,138,174,173,93,128,78,217,218,93,209,181,181,11,208,41,91,187,43,186,182,118,1,58,101,107,215,150,58,11,61,69,63,227,48,208,155,208,115,244,70,66,159,132,158,163,79,36,116,21,41,64,39,21,105,85,145,2,116,82,145,86,21,41,64,39,21,105,85,145,2,116,82,145,86,21,41,64,39,21,105,85,145,2,116,82,145,86,21,41,64,39,21,105,168,72,115,244,32,21,105,168,72,1,58,169,72,67,69,10,208,73,69,26,42,82,128,78,42,210,80,145,2,116,82,145,134,138,20,160,147,138,52,84,164,0,157,84,164,161,34,5,232,164,34,13,21,41,64,39,21,105,168,72,1,58,169,72,139,138,52,71,47,164,34,45,42,82,128,78,42,210,162,34,5,232,164,34,45,42,82,128,78,42,210,162,34,5,232,164,34,213,89,187,8,157,84,164,58,107,23,161,147,138,84,103,237,34,116,82,145,234,172,93,132,78,42,82,157,181,139,208,73,69,234,42,210,28,221,73,69,234,42,82,128,78,42,82,87,145,2,116,82,145,186,138,20,160,147,138,212,85,164,0,157,84,164,174,34,5,232,164,34,117,21,41,64,39,21,169,171,72,1,58,169,72,93,69,10,208,73,69,234,42,82,128,78,42,82,83,145,230,232,70,42,82,83,145,2,116,82,145,154,138,20,160,147,138,212,84,164,0,157,84,164,166,34,5,232,164,34,53,21,41,64,39,21,169,169,72,1,58,169,72,77,69,10,208,73,69,106,42,82,128,78,42,210,109,252,62,135,97,118,238,240,225,209,55,156,183,70,95,139,107,152,133,158,160,79,23,28,6,122,19,122,142,222,72,232,147,208,115,244,137,132,62,10,61,71,31,73,232,131,208,115,244,129,132,94,133,158,163,87,18,122,8,61,71,15,18,122,17,122,142,94,72,232,46,244,28,221,73,232,42,82,128,78,42,210,170,34,205,209,43,169,72,171,138,20,160,147,138,180,170,72,1,58,169,72,171,138,20,160,147,138,180,170,72,1,58,169,72,107,93,220,244,37,70,130,190,225,48,208,67,232,57,122,144,208,139,208,115,244,66,66,119,161,231,232,78,66,55,161,231,232,198,65,143,89,232,41,250,25,135,129,222,132,158,163,55,18,250,36,244,28,125,34,161,143,66,207,209,71,18,250,32,244,28,125,32,161,171,72,1,58,169,72,67,69,10,208,73,69,26,42,82,128,78,42,210,80,145,2,116,82,145,134,138,20,160,147,138,180,168,72,115,244,66,42,210,162,34,5,232,164,34,45,42,82,128,78,42,210,50,44,117,126,151,175,235,254,5,189,243,106,255,5,125,195,121,65,255,241,7,82,55,132,8,100,173,0,0,13,10,48,13,10,13,10]}}
2018-07-26T11:29:48.247Z express-expeditious GET /api/anlage/fahrt/39/05.06.2018%2011:25:20/120 - etag did not match. returning cached data.
2018-07-26T11:29:48.248Z express-expeditious GET /api/anlage/fahrt/39/05.06.2018%2011:25:20/120 - cache error. proceed with request TypeError: Cannot read property 'write' of null
    at onCacheReturned (I:\Projekt\BDEDash\node_modules\express-expeditious\lib\middleware.js:107:22)
    at I:\Projekt\BDEDash\node_modules\safejson\src\index.js:53:14
    at deferred (I:\Projekt\BDEDash\node_modules\safejson\src\index.js:15:5)
    at Object.exports.parse (I:\Projekt\BDEDash\node_modules\safejson\src\index.js:51:3)
    at onGetCallComplete (I:\Projekt\BDEDash\node_modules\expeditious\lib\handlers.js:33:18)
    at Immediate._onImmediate (I:\Projekt\BDEDash\node_modules\expeditious-engine-memory\expeditious-memory-engine.js:50:9)
    at runCallback (timers.js:696:18)
    at tryOnImmediate (timers.js:667:5)
    at processImmediate (timers.js:649:5)
2018-07-26T11:29:48.248Z express-expeditious GET /api/anlage/fahrt/39/05.06.2018%2011:25:20/120 - request will be used to build a cache
I:\Projekt\BDEDash\node_modules\express-expeditious\lib\middleware.js:254
        _write = res.socket.write.bind(res.socket)
                            ^

TypeError: Cannot read property 'write' of null
    at processRequest (I:\Projekt\BDEDash\node_modules\express-expeditious\lib\middleware.js:254:29)
    at onCacheReturned (I:\Projekt\BDEDash\node_modules\express-expeditious\lib\middleware.js:89:9)
    at I:\Projekt\BDEDash\node_modules\safejson\src\index.js:55:14
    at deferred (I:\Projekt\BDEDash\node_modules\safejson\src\index.js:15:5)
    at Object.exports.parse (I:\Projekt\BDEDash\node_modules\safejson\src\index.js:51:3)
    at onGetCallComplete (I:\Projekt\BDEDash\node_modules\expeditious\lib\handlers.js:33:18)
    at Immediate.<anonymous> (I:\Projekt\BDEDash\node_modules\expeditious-engine-memory\expeditious-memory-engine.js:50:9)
    at runCallback (timers.js:696:18)
    at tryOnImmediate (timers.js:667:5)
    at processImmediate (timers.js:649:5)`

**Comment by evanshortiss** [comment]
@fupps can you incldue your Node.js version and the version of express-exeditious you are using? Did you get a solution?

**Comment by evanshortiss** [comment]
Also can you include some sample code? That reproduces the issue?

**Comment by evanshortiss** [comment]
Closing, but please reopen if you have more information or are still seeing the error since I'd love to determine if it's something we need to check for and patch
